### DISCUSSION TITLE
Questions running pypiserver with gunicorn and tornado

### DISCUSSION THREAD

**Discussion by jfacevedo** [pypiserver/pypiserver#605] [General]
I am currently running pypiserver v1.2.2 with tornado in a productive environment and I need to update it. 

I currently run pypiserver (v.1.2.2) as follows and it works fine:
`pypi-server --server tornado -p 8080 -P . -a . /home/pypi`

I am testing the latest version of pypiserver (v2.2.0) and I see that some things have changed at runtime.

I find a lot of slowness when running pypiserver as:
`pypi-server run -p 8080 -P . -a . /home/pypi`

I am trying gunicorn as:
`'gunicorn -w4 -b 127.0.0.1:8080 'pypiserver:app(roots=“/home/pypi”)'`

And tornado using bottle as:
`python3 -m bottle --server tornado -b 127.0.0.1.1:8080 'pypiserver1:app(roots=“/home/pypi”)'`

But in both cases I can't get the pypyserver home page to show (attached screenshot). Which is shown when running pypiserver as in the first option.

Can someone give me a hand and tell me what I should change or adjust to run pypiserver with gunicorn or tornado and can show the pypiserver home page, please?

![Screenshot 2024-09-17 at 10 51 20 AM](https://github.com/user-attachments/assets/c0b9693b-8850-4f75-9eb3-23d8016a47b9)

**Comment by dee-me-tree-or-love** [comment]
Hello @jfacevedo, sorry once again for my slow responses on the discussions here! 
I'm sorry that you have been experiencing issues running the newer `pypi-server`. 

Seeing that you have closed this discussion, may I ask whether you found a solution? 
If you are willing to share, I'd be really grateful (and maybe it can help some new readers in this thread)! 

If you are still experiencing any troubles please let me know, unfortunately my response rate is indeed not the highest 😞 but I'll be happy to think along. 🙏
